common/reducers/message.ts

```diff
@@ -0,0 +1,26 @@
+import { MessageAction, SignLocalMessageSucceededAction, TypeKeys } from 'actions/message';
+import { ISignedMessage } from 'libs/signing';
+
+export interface State {
+  signed?: ISignedMessage | null;
```

Comment:
Would love to see an `isSigning` boolean here, and stop users from trying to sign again / providing a loader while waiting for signature.

Reply:
:+1: not strictly necessary for the QR codes since the modal takes over, but I can see how it can be a thing with hardware wallets.